src/liballoc/collections/btree/node.rs

```diff
@@ -83,8 +83,8 @@ struct LeafNode<K, V> {
 
     /// The arrays storing the actual data of the node. Only the first `len` elements of each
     /// array are initialized and valid.
-    keys: [K; CAPACITY],
-    vals: [V; CAPACITY],
+    keys: MaybeUninit<[K; CAPACITY]>,
+    vals: MaybeUninit<[V; CAPACITY]>,
```

Comment:
Based on the comment above, should this instead be `[MaybeUninit<V>; CAPACITY]`?

(That might be a pain, though, and this is no worse than before, so might not be this PR.)

Reply:
That seems like a pretty equivalent type to me. Which comment above?

Reply:
Should we document that you can partially-initialize `MaybeUninit`, i.e. that it "distributes over products"?

Reply:
I guess this is forth saying, yes.

Unfortunately actually implementing this is not easy, same problem as for `Cell` (which also distributes over products).